image-loader/app/lib/imaging/FileMetadataReader.scala

```diff
@@ -107,24 +110,31 @@ object FileMetadataReader {
     } getOrElse Map()
 
   private val datePattern = "(.*[Dd]ate.*)".r
-  private def exportXmpProperties(metadata: Metadata, imageId:String): Map[String, String] =
-    Option(metadata.getFirstDirectoryOfType(classOf[XmpDirectory])) map { directory =>
-      directory.getXmpProperties.asScala.toMap.mapValues(nonEmptyTrimmed).collect {
-        case (datePattern(key), Some(value)) => key -> ImageMetadataConverter.cleanDate(value, key, imageId)
-        case (key, Some(value)) => key -> value
-      }
-    } getOrElse Map()
+
+  private def xmpDirectoryToMap(directory: XmpDirectory, imageId: String): Map[String, String] = {
+    directory.getXmpProperties.asScala.toMap.mapValues(nonEmptyTrimmed).collect {
+      case (datePattern(key), Some(value)) => key -> ImageMetadataConverter.cleanDate(value, key, imageId)
+      case (key, Some(value)) => key -> value
+    }
+  }
+
+  private def exportXmpProperties(metadata: Metadata, imageId:String): Map[String, String] = {
+    val directories = metadata.getDirectoriesOfType(classOf[XmpDirectory]).asScala.toList
```

Comment:
this is the main change, going from `getFirstDirectoryOfType` to `getDirectoriesOfType`